File: skill_sdk/tests/fixtures/ts-project/src/shop/payment/PayPalGateway.ts
import { PaymentGateway } from './PaymentGateway';
import { PaymentResult } from './PaymentResult';

/**
 * PayPal-backed payment gateway.
 */
export class PayPalGateway implements PaymentGateway {
  constructor(
    private readonly clientId: string,
    private readonly clientSecret: string
  ) {}

  charge(amount: number, currency: string): PaymentResult {
    const txId = `pp_ch_${this.hashFloat(amount)}`;
    return new PaymentResult(true, txId, `PayPal charge ${amount} ${currency}`);
  }

  refund(transactionId: string): PaymentResult {
    return new PaymentResult(
      true,
      `pp_ref_${transactionId}`,
      `PayPal refund ${transactionId}`
    );
  }

  private hashFloat(f: number): number {
    return Math.floor(f * 100);
  }
}

File: skill_sdk/tests/fixtures/java-project/src/main/java/com/fixture/shop/payment/PayPalGateway.java
package com.fixture.shop.payment;

/**
 * PayPal-backed payment gateway.
 */
public class PayPalGateway implements PaymentGateway {

    private final String clientId;
    private final String clientSecret;

    public PayPalGateway(String clientId, String clientSecret) {
        this.clientId = clientId;
        this.clientSecret = clientSecret;
    }

    @Override
    public PaymentResult charge(double amount, String currency) {
        String txId = "pp_ch_" + hashFloat(amount);
        return new PaymentResult(true, txId, "PayPal charge " + amount + " " + currency);
    }

    @Override
    public PaymentResult refund(String transactionId) {
        return new PaymentResult(true, "pp_ref_" + transactionId,
                "PayPal refund " + transactionId);
    }

    private static int hashFloat(double f) {
        return (int) (f * 100);
    }
}
